### ISSUE TITLE
Implement exporting an image sequence from the AnimationPlayer editor

### ISSUE THREAD

**Issue by erayzesen** [godotengine/godot-proposals#3265]
### Describe the project you are working on

For my game, I make cut scenes and in-game trailer logo animations in the game engine. Godot does a pretty good job at this. I don't need any other software for these and being able to handle these things with Godot is an advantage for me. 

### Describe the problem or limitation you are having in your project

When my game is finished and I am going to launch, I need to export these beautiful animations that I have made with AnimationPlayer as native. The hard part is making the animations anyway, it's weird that I can't export them in at least one visual format. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

When most of us finish our games, we want to use the animation works (cutscene, intro, logo animation...etc) we perform in the game engine in in-game launch images and trailers. Godot is very talented for in-game animations, but the animations we make stay inside godot. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

As the simplest solution, the "Export" option can be added to the "Edit" drop-down menu in the Animation panel. Format and various settings can be added from the panel that opens when we press the "Export" button. Even just the png output may be sufficient, in the form of "Export PNG". 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think this feature will be used frequently. But even if we assume that it will not be used;

It cannot be said to be practical, but of course it can be solved with a script. Of course, in this scenario, a developer is sitting at the head of godot to do this job. We want teams to use Godot, what if someone on the creative side of the team wants to quickly export these animations? Generally, these requests are discussed in user scenarios as "a software developer's use of godot". But an artist, not a software developer, can take over this job. 

### Is there a reason why this should be core and not an add-on in the asset library?

You can do this with plugins, you can do this by writing a script and running the game. But none of them will be as fast and useful as the built-in support of the game engine in this regard.

Said request may be very simple under the hood yes, but it will make an incredible difference. Godot will be able to export options the animation features it already has and users will be able to use this for the launch of their games and for different purposes. 

In other words, Godot will have crowned the extensive features it has with AnimationPlayer with a simple image export option.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/3083.

The issue with adding this directly to AnimationPlayer is that you often want some scripts to run at the same time. Besides, you'll also want the project to run to make sure the viewport size can be set correctly. So I'd prefer we expose just the low-level building blocks (video recording) and let the users write an add-on to use it in a more convenient manner.

**Comment by erayzesen** [comment]
" you often want some scripts to run at the same time."

Scripts do not need to be run, it is enough if the movements we make in the editor in animationplayer are exported visually.

In the scripts we make, we actually get the image from the viewport. In the scenario I'm talking about, what if it takes the image from the first Viewport above the AnimationPlayer while the "export" command is being executed?

All I'm requesting here is to export the image we see in the scene (possible first top Viewport) when we hit play in the AnimationPlayer. 

I don't know if a extension is possible that can do this within the editor. In the current scenario, we can only run the game and save it from the viewport.  For this, we will have to use the script in question in the relevant places, adjust it, run the game and get the images.

**Comment by Calinou** [comment]
> It will also bring along small jumps caused by fps drops during playback.

This is something that can be fixed by non-realtime simulation (`--fixed-fps N` command line argument).

**Comment by erayzesen** [comment]
> > It will also bring along small jumps caused by fps drops during playback.
> 
> This is something that can be fixed by non-realtime simulation (`--fixed-fps N` command line argument).

What if I want to save the animation as transparent png in the application you mentioned? So this is not a very far scenario, I am publishing my game in a market, I want to export the logo animation that I have carefully prepared in the game as a transparent png, I have no choice but to realize this with interesting scripts.

**Comment by Calinou** [comment]
> What if I want to save the animation as transparent png in the application you mentioned?

It's possible to use a transparent viewport (or even make the root viewport transparent by setting its `transparent` property to `true`), then save its contents using `Image.save_png()`. See https://github.com/Calinou/godot-video-rendering-demo for an example of this.

**Comment by erayzesen** [comment]
> > What if I want to save the animation as transparent png in the application you mentioned?
> 
> It's possible to use a transparent viewport (or even make the root viewport transparent by setting its `transparent` property to `true`), then save its contents using `Image.save_png()`. See https://github.com/Calinou/godot-video-rendering-demo for an example of this.

I know how to do this with script. I wonder if there could be an option in the editor instead of having to do this for every animation I want to save. A similar question was asked in 2017 before this proposal section was made;

https://github.com/godotengine/godot/issues/12240

**Comment by Calinou** [comment]
Someone is working on an add-on for this: https://github.com/AhmedElTabarani/AnimationRecorder

**Comment by Calinou** [comment]
This is implemented as of Godot 4.0 with [Movie Maker mode](https://docs.godotengine.org/en/latest/tutorials/animation/creating_movies.html), which works in both 2D and 3D.

**Comment by erayzesen** [comment]
@Calinou  I know all these features added on 4.0 in a magical way.  But thank you for your helps in this issue.
